x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractorFactory.java

```diff
@@ -99,10 +101,13 @@ public DataFrameDataExtractor newExtractor(boolean includeSource) {
     public static void create(Client client,
                               DataFrameAnalyticsConfig config,
                               ActionListener<DataFrameDataExtractorFactory> listener) {
-        validateIndexAndExtractFields(client, config.getHeaders(), config.getDest(), config.getAnalysesFields(), ActionListener.wrap(
-            extractedFields -> listener.onResponse(
-                new DataFrameDataExtractorFactory(client, config.getId(), config.getDest(), extractedFields, config.getHeaders())),
-            listener::onFailure
+        List<DataFrameAnalysis> analyses = DataFrameAnalysesUtils.readAnalyses(config.getAnalyses());
```

Comment:
We could extract those 2 lines in a method `resolveResultFields` and reuse it further down.